src/framework/components/element/text-element.js

```diff
@@ -900,9 +931,7 @@ class TextElement {
                     _xMinusTrailingWhitespace = _x;
                 }
 
-                var isWordBoundary = WORD_BOUNDARY_CHAR.test(char);
-                var isCJK = CJK_CHAR.test(char);
-                if (isWordBoundary || isCJK) { // char is space, tab, or dash OR CJK
+                if (isWordBoundary() || (isValidNextChar() && (isNextCJKBoundary() || isNextCJKWholeWord()))) {
```

Comment:
I find this `if` statement way easier to read now - nice. But personally, I would define the 4 functions at module scope near the top instead of inside _updateMeshes (where they'll be allocated on every call). They'd just need the char passing as an argument.

Reply:
makes sense - moved functions to module scope